extensions/0znzw/0tils.js

```diff
@@ -0,0 +1,1577 @@
+    sprite = sprite || "";
+    const target = _SGCUtils.ids.getTarget(sprite);
+    return vm.getVariableValue(target.id, _SGCUtils.ids.getVariableByName(name, "")
+      .id);
+  };
+  _SGCUtils.setVariableByName = function(name, value, sprite) {
+    sprite = sprite || "";
+    const target = _SGCUtils.ids.getTarget(sprite);
+    return vm.setVariableValue(target.id, _SGCUtils.ids.getVariableByName(name, "")
+      .id, value);
+  };
+  const stageVariables = [{
+    value: 0
+  }];
+  const zips = [];
+  const stageVariablesDisplay = ['my variable'];
+
+  const allowDuplicates = !urlParams.has("0tils-enableDuplicates");
+  const allowUnsafe = !urlParams.has("0tils-allowUnsafe");
+  const showParenSpam = urlParams.has("0tils-showParenSpam");
+  const showLabels = !urlParams.has("0tils-showExtraLabels");
+  
+  function addLPlabel(text) {
+    if (showParenSpam) return `(${text}) `;
+    return '';
+  }
+  
+  class SGCUtils {
+    getInfo() {
+      return {
+        id: 'SGCUtils',
+        name: '0 Utilities',
+        menuIconURI: icon,
+        docsURI: 'https://survexe1pc.github.io/0tils',
+        color1: '#BE3DC0', // block color
+        color2: '#D882EE', // border+dropdown color
+        color3: '#D882EE', // input color
+        blocks: [{
+            hideFromPalette: showLabels,
+            opcode: 'getVersion',
+            blockType: Scratch.BlockType.REPORTER,
+            text: '0tils version'
+          }, {
+            opcode: "emptyFunctionForLabels",
+            blockType: "label",
+            text: "Encryption",
+            hideFromPalette: showLabels
+          }, {
+          opcode: 'doAES',
+          blockType: Scratch.BlockType.REPORTER,
+          text: addLPlabel("AES")+'[mode] [data] with password: [pwd] and [bytes] bytes.',
+          arguments: {
+            mode: {
+              type: Scratch.ArgumentType.STRING,
+              menu: 'cryptmodes'
+            },
+            pwd: {
+              type: Scratch.ArgumentType.STRING,
+              defaultValue: 'password'
+            },
+            data: {
+              type: Scratch.ArgumentType.STRING,
+              defaultValue: 'Hello, World!'
+            },
+            bytes: {
+              type: Scratch.ArgumentType.STRING,
+              menu: 'aesbytes'
+            }
+          }
+        }, {
+          hideFromPalette: allowUnsafe,
+          opcode: 'doRSA',
+          blockType: Scratch.BlockType.REPORTER,
+          text: addLPlabel("RSA")+'[mode] [data] with key: [key].',
+          arguments: {
+            mode: {
+              type: Scratch.ArgumentType.STRING,
+              menu: 'cryptmodes'
+            },
+            key: {
+              type: Scratch.ArgumentType.STRING,
+              defaultValue: ''
+            },
+            data: {
+              type: Scratch.ArgumentType.STRING,
+              defaultValue: 'Hello, World!'
+            }
+          }
+        }, {
+          opcode: "emptyFunctionForLabels",
```

Comment:
opcode on label doesn't do anything by the way. just don't list it

Reply:
will do

Reply:
fixed